src/library/scala/Option.scala

```diff
@@ -26,8 +26,19 @@ object Option {
    *
    *  @param  x the value
    *  @return   Some(value) if value != null, None if value == null
+   *  @deprecated if you intend to filter out null, use [[whenNonNull]] instead,
+   *              as its meaning is clearer; otherwise use [[Some.apply Some]]
    */
-  def apply[A](x: A): Option[A] = if (x == null) None else Some(x)
+  @deprecated("Use `whenNonNull` or `Some.apply` instead", since = "2.13.0")
+  @inline def apply[A](x: A): Option[A] = whenNonNull(x)
+
+  /** An Option factory which creates Some(x) if the argument is not null,
+    *  and None if it is null.
+    *
+    *  @param  a the value
+    *  @return   Some(value) if value != null, None if value == null
+    */
+  def whenNonNull[A](a: A): Option[A] = if (a == null) None else Some(a)
```

Comment:
Why not just add `of`/`ofNonNull` and `ofNullable` just as the Java one?

Reply:
https://github.com/vavr-io/vavr/blob/41cd78ff44ee30e9c258053ea21aeaadb1c8e936/vavr-control/src/main/java/io/vavr/control/Option.java#L57